### PR TITLE
Scrolling fixes for desktop & mobile

### PR DESCRIPTION
**Pull request by dozoisch**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ts-simditor-mention",
-  "version": "4.0.0-alpha.1",
+  "version": "4.0.1",
```
Comment (gaurav-townsqd) [inline bower.json]: We should bump it to `4.0.0-alpha.2` instead.

The idea is to work towards a stable `4.0.0` release by incrementally improving what this library does. That `4.0.0` release will be our gift to the community. Until then, every version is alpha.

File: package.json [hunk 2]
```diff
@@ -22,10 +22,12 @@
     "simditor": "2.3.x"
   },
   "scripts": {
-    "test": "grunt test --verbose"
+    "test": "grunt test --verbose",
+    "build": "grunt coffee umd --verbose"
```
Comment (gaurav-townsqd) [inline package.json:26]: Nice 👍

File: src/simditor-mention.coffee [hunk 3]
```diff
@@ -224,17 +230,33 @@ class SimditorMention extends SimpleModule
...
+        touches = e.originalEvent.targetTouches[0]
+        @popoverStartY = touches.clientY
+        @popoverIsTouching = true
+        @popoverCancelTouch = false
+      undefined
+    @popoverEl.on 'touchmove', '.item', (e)=>
+      if @popoverIsTouching
+        touches = e.originalEvent.targetTouches[0]
+        currentY = touches.clientY
+        @popoverCancelTouch = @popoverCancelTouch || (Math.abs(@popoverStartY - currentY) > 15)
```
Comment (gaurav-townsqd) [inline src/simditor-mention.coffee]: Is `15` just an arbitrary small number, or was it chosen based on something specific (like some element's height) ?

File: src/simditor-mention.coffee [hunk 4]
```diff
@@ -224,17 +230,33 @@ class SimditorMention extends SimpleModule
...
+      undefined
+
+    @popoverEl.on 'touchend', '.item', (e)=>
+      @popoverIsTouching = false;
+      if !@popoverCancelTouch
+        @selectItem()
```
Comment (gaurav-townsqd) [inline src/simditor-mention.coffee:258]: I didn't realize that this was an issue :(  Thanks for fixing it!

File: lib/simditor-mention.js [hunk 5]
```diff
@@ -280,19 +285,50 @@ SimditorMention = (function(superClass) {
...
+    this.popoverEl.on('mousedown', '.item', (function(_this) {
       return function(e) {
         _this.selectItem();
         return false;
       };
     })(this));
-    return $itemsEl.on('mousewheel', function(e, delta) {
-      $(this).scrollTop($(this).scrollTop() - 10 * delta);
-      return false;
-    });
```
Comment (gaurav-townsqd) [inline lib/simditor-mention.js:295]: Craziness. Thanks for fixing this!

**Review by gaurav-townsqd — COMMENTED** [review]
Just 1 comment about the versioning. Everything else looks good.

File: src/simditor-mention.coffee (hunk 3 above)
Comment (dozoisch) [reply src/simditor-mention.coffee]: It's an arbitrary number based of when I implemented something similar. Goal is to let user move a bit on the screen without cancelling but if they scroll enough we cancel. We can try it on device and tweak it accordingly afterwards!

Comment (dozoisch) [reply src/simditor-mention.coffee]: I should actually lower it to 10. 10 is the same used in the tap event plugin for react

https://github.com/zilverline/react-tap-event-plugin/blob/master/src/TapEventPlugin.js#L49

Comment (gaurav-townsqd) [reply src/simditor-mention.coffee]: @dozoisch I like the idea of assigning a variable to this number so the intent is not lost :)

File: bower.json (hunk 1 above)
Comment (dozoisch) [reply bower.json]: Alright! Yeah I was thinking that with the decaffeination, I'm not sure we could actually merge it back. We can always hold off of decaffeination

File: lib/simditor-mention.js (hunk 5 above)
Comment (dozoisch) [reply lib/simditor-mention.js:295]: Yeah don't know why it's trying to hijack scroll behavior!

**Review by gaurav-townsqd — APPROVED** [review]
LGTM. Merge away!

**Pull request merged**